tests/TestCase/ORM/QueryTest.php

```diff
@@ -4265,4 +4266,69 @@ public function testSubqueryJoinClause()
         $this->assertEquals(1, $results[0]->id);
         $this->assertEquals(2, $results[0]->total_articles);
     }
+
+    /**
+     * Tests that queries that fetch associated data in separate queries do properly
+     * inherit the hydration and results casting mode of the parent query.
+     *
+     * @return void
+     */
+    public function testSelectLoaderAssociationsInheritHydrationAndResultsCastingMode()
+    {
+        $articles = $this->getTableLocator()->get('Articles');
+
+        $tags = $articles->belongsToMany('Tags');
+        $tags->belongsToMany('Articles');
+
+        $comments = $articles->hasMany('Comments');
+        $comments
+            ->belongsTo('Articles')
+            ->setStrategy(BelongsTo::STRATEGY_SELECT);
+
+        $disabled = true;
+        $enabled = true;
+
+        $articles
+            ->find()
+            ->contain('Comments', function (Query $query) use (&$disabled) {
+                $disabled =
+                    $disabled &&
+                    !$query->isHydrationEnabled() &&
+                    !$query->isResultsCastingEnabled();
```

Comment:
You can use assertions inside these closures to avoid needing to close over disabled and enabled.

Reply:
It's a habit... I'm always thinking, "_what if a closure isn't being invoked for some reason_", but in this case it wouldn't prevent things from not being covered anyways, as the initial values equal the expected values 🙃